pwiz_tools/Skyline/Model/FullScanProperties.cs

```diff
@@ -0,0 +1,141 @@
+using System.Linq;
+using System.Resources;
+using pwiz.ProteowizardWrapper;
+using pwiz.Skyline.Model.Hibernate;
+using pwiz.Skyline.Util.Extensions;
+
+namespace pwiz.Skyline.Model
+{
+    public class FullScanProperties : GlobalizedObject
+    {
+        [TypeConverter(typeof(ExpandableObjectConverter))]
+        public class InstrumentInfo : GlobalizedObject
+        {
+            [TypeConverter(typeof(ExpandableObjectConverter))]
+            public class InstrumentComponentsInfo : GlobalizedObject
+            {
+                protected override ResourceManager GetResourceManager()
+                {
+                    return FullScanPropertiesRes.ResourceManager;
+                }
+                [NotifyParentProperty(true)][Category("InstrumentInfo")] public string Ionization { get; set; }
+                [NotifyParentProperty(true)][Category("InstrumentInfo")] public string Analyzer { get; set; }
+                [NotifyParentProperty(true)][Category("InstrumentInfo")] public string Detector { get; set; }
+
+                public override string ToString()
+                {
+                    return Analyzer;
+                }
+            }
+
+            protected override ResourceManager GetResourceManager()
+            {
+                return FullScanPropertiesRes.ResourceManager;
+            }
+            [NotifyParentProperty(true)][Category("InstrumentInfo")] public string InstrumentSerialNumber { get; set; }
+            [NotifyParentProperty(true)][Category("InstrumentInfo")] public string InstrumentModel { get; set; }
+            [NotifyParentProperty(true)][Category("InstrumentInfo")] public string InstrumentManufacturer { get; set; }
+            [NotifyParentProperty(true)][Category("InstrumentInfo")] public InstrumentComponentsInfo InstrumentComponents { get; set; }
+
+            public override string ToString()
+            {
+                if(InstrumentModel != null || InstrumentManufacturer != null)
+                    return TextUtil.SpaceSeparate(InstrumentManufacturer, InstrumentModel);
+                return base.ToString();
+            }
+        } 
+        protected override ResourceManager GetResourceManager()
+        {
+            return FullScanPropertiesRes.ResourceManager;
+        }
+
+        public static FullScanProperties CreateProperties(MsDataSpectrum spectrum)
+        {
+            if(spectrum == null)
+                return null;
+            var res = new FullScanProperties();
+            res.FilePath = spectrum.SourceFilePath;
+            res.FileName = Path.GetFileName(spectrum.SourceFilePath);
+            if (spectrum.PrecursorsByMsLevel.Any())
+            {
+                var precursor = spectrum.Precursors.FirstOrDefault();
+                res.PrecursorMz = precursor.PrecursorMz.HasValue ? precursor.PrecursorMz.Value.Value.ToString(Formats.Mz) : null;
+                res.Charge = spectrum.NegativeCharge ? @"-" : @"+" + precursor.ChargeState;
```

Comment:
I noticed "precursor.ChargeState" is actually always null.
The following line of code should to be added to "MsDataFileImpl.CreatePrecursor":
`ChargeState = GetChargeStateValue(p),
`

Reply:
This change actually breaks the DdaSearchTest. The number of proteins changes, so the assertion fails.

Reply:
Oops. It seems like we must have never been passing charge states in the raw data through to MSAmanda. That would definitely change the results (and it should be better). Can you update the numbers? Use IsRecordMode => true to get all the numbers so it doesn't just fail at the first assert. Or I can do it.

Reply:
I think I revert it for now.

Reply:
Yes, sure.